atlp-rwanda/script-heroes-bn-backend
# #174797543 Book an Accommodation

### What does this PR do?
Allow users to book an accomodation

### Description of Task to be completed?

1. Pull the repo to get the project's latest version
2. Create a branch to work on
3. Select accommodation from the database.
4. Change` isBooked`  to be true and `userId` to be the id of a user who booked accommodation.
5. Validate User's Input
6. Write Tests
7. Add, commit, and push changes to GitHub

### How should this be manually tested?
Clone a repo, run `npm run dev` to run the project. After having a project running on a port using `localhost:PORTNO` to run the project in the browser. Go to `http://localhost:PORTNO/api/accommodation/book` . Make sure that you have `accomodationId` and `roomId` in the request's body. After sending the request, the message of success and the requester's information will be returned.
### Any background context you want to provide?
The request's body should look like this:
`{
 "accommodationId":2,
"roomId": 3,
    "checkInDate": "January 1, 2020",
    "checkOutDate": "Feb 2, 2020"
}`
### What are the relevant pivotal tracker stories?
[#174797543](https://www.pivotaltracker.com/n/projects/2465784)
